core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -2456,15 +2457,62 @@ private RecordCursorFactory generateJoins(QueryModel model, SqlExecutionContext
                                             );
                                         }
                                     } else {
-                                        if (slave.supportsTimeFrameCursor()) {
-                                            master = new AsOfJoinNoKeyFastRecordCursorFactory(
-                                                    configuration,
-                                                    createJoinMetadata(masterAlias, masterMetadata, slaveModel.getName(), slaveMetadata),
-                                                    master,
-                                                    slave,
-                                                    masterMetadata.getColumnCount()
-                                            );
-                                        } else {
+                                        boolean created = false;
+                                        if (fastAsOfJoins) {
+                                            if (slave.supportsTimeFrameCursor()) {
+                                                master = new AsOfJoinNoKeyFastRecordCursorFactory(
+                                                        configuration,
+                                                        createJoinMetadata(masterAlias, masterMetadata, slaveModel.getName(), slaveMetadata),
+                                                        master,
+                                                        slave,
+                                                        masterMetadata.getColumnCount()
+                                                );
+                                                created = true;
+                                            }
+
+                                            if (!created && slave.supportsFilterStealing() && slave.getBaseFactory().supportsTimeFrameCursor()) {
+                                                RecordCursorFactory slaveBase = slave.getBaseFactory();
+                                                Function stolenFilter = slave.getFilter();
```

Comment:
We need to close compiled filter, bind var memory, and bind var functions here and before the second `FilteredAsOfJoinNoKeyFastRecordCursorFactory` constructor call. Otherwise, we'll have a memory leak. Alternatively, we could use them (see the near-by suggestion).

Could you add some tests around JIT filters + bind vars, so that our mem leak detection kicks in?